src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsParsing.kt

```diff
@@ -0,0 +1,97 @@
+package uk.gov.dwp.dataworks.aws
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.PropertyNamingStrategy
+import com.fasterxml.jackson.databind.cfg.MapperConfig
+import com.fasterxml.jackson.databind.introspect.AnnotatedField
+import com.fasterxml.jackson.databind.introspect.AnnotatedMethod
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.core.io.Resource
+import org.springframework.stereotype.Component
+import uk.gov.dwp.dataworks.AwsIamPolicyJsonObject
+import uk.gov.dwp.dataworks.services.ConfigurationResolver
+import java.lang.reflect.Modifier
+
+@Component
+class AwsParsing(){
+    @Autowired
+    private lateinit var  configurationResolver: ConfigurationResolver
+
+    @Autowired
+    private lateinit var awsCommunicator: AwsCommunicator
+
+    /**
+     * Helper method to initialise the lateinit vars [taskRolePolicyString] and [jupyterBucketAccessRolePolicyString] by
+     * converting the JSON to an instance of `JsonObject` data class and adding any requirements before serialising back to JSON
+     *
+     * @return [taskRolePolicyString] and [jupyterBucketAccessRolePolicyString] respectively.
+     */
+    fun parsePolicyDocument(resource: Resource, sidAndAdditions: Map<String, List<String>>, key: String): String{
+        val mapper = ObjectMapper()
+                .setPropertyNamingStrategy(AwsPropertyNamingStrategy())
+        val obj = mapper.readValue(resource.url, AwsIamPolicyJsonObject::class.java)
```

Comment:
I think we get nicer error messages from using a `File` object as opposed to `URL`. For example if the file is not found you get `FileNotFoundException` but with URL that would be wrapped as a more generic `IOException`